### PR TITLE
Clean up navigation logic (1/x).

### PR DESCRIPTION
**Pull request by chrisbobbe**
This is the first PR of several mostly NFC PRs I'd like to make, to make our navigation logic better-structured and easier to work with, now that we're up-to-date with the latest version of React Navigation (v5; that was #4296). I expect #4417 to be in this path. Eventually, I expect #3066 will be much easier to fix according to react-navigation's [official guide](https://reactnavigation.org/docs/handling-safe-area) for supporting safe areas.

This PR starts off that series of PRs by making our naming conventions a bit more consistent.

### CODE REVIEW

**Comment by chrisbobbe** [comment]
Toward the end of this branch, I hint at wanting to make a new stack navigator for what's on the settings tab, on the ["stack navigator for each tab"](https://reactnavigation.org/docs/tab-based-navigation#a-stack-navigator-for-each-tab) pattern, and putting that tab's "sub"-screens (the language-picker screen, etc.) on that. I think this would better organize our code. It would also mean that the tab bar will be visible even while looking at screens under "settings", like the language-picker screen. I'm happy to discuss this more, perhaps on CZO, if this sounds like something we might not want to do. Some useful background (apart from the doc I've just linked to) can be found

- [here](https://reactnavigation.org/docs/hiding-tabbar-in-screens/), for a react-navigation pattern that looks like it may have informed the current code
- [here](https://reactnavigation.org/docs/nesting-navigators#best-practices-when-nesting), for a react-navigation warning against excessively nesting navigators, which I think shouldn't deter us from making the change
- [here](https://material.io/components/bottom-navigation#behavior) (specifically, the "On Android" and "On iOS" bullets under "Behavior" > "Navigation"), for an iOS design recommendation from Material Design that would be better satisfied after this change, while the change doesn't prevent us from satisfying a (different) recommendation for Android (i.e., we could use [`unmountOnBlur`](https://reactnavigation.org/docs/bottom-tab-navigator#unmountonblur) for the settings-tab screen on Android).

**Comment by chrisbobbe** [comment]
(Just fixed some small conflicts.)

**Review by gnprice — COMMENTED** [review]
Thanks! This generally all looks great; one comment below on the last commit.

I'll take a look also at those links you mention in the comment.

File: src/settings/SettingsMenuScreen.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ type Props = $ReadOnly<{|
   dispatch: Dispatch,
 |}>;
 
-class SettingsScreen extends PureComponent<Props> {
+class SettingsMenuScreen extends PureComponent<Props> {
```
[the remark this answers is not in the document]
Comment (chrisbobbe) [reply src/settings/SettingsMenuScreen.js:43]: Mm, good point.

> How about "SettingsInitialScreen"? Or "SettingsRootScreen", or "SettingsMainScreen"?

I like it! Maybe `SettingsInitialScreen`.

**Comment by gnprice** [comment]
> I think this would better organize our code. It would also mean that the tab bar will be visible even while looking at screens under "settings", like the language-picker screen.

Hmm. I'm not sure I've seen this behavior on any other app. It seems like it could make the UI more crowded and busy than it needs to be.

Is there an example of another app where you've seen this behavior and like the effect?

The "Nesting navigators" doc you link to above offers this thought, which seems right to me:
> Think of nesting navigators as a way to achieve the UI you want rather than a way to organize your code.

I think for organizing our code we can find other ways that don't show themselves in the UI. (One such way is suggested just after that in that doc.) Happy to discuss in more detail on CZO.

**Comment by chrisbobbe** [comment]
Thanks for the review! We've discussed the "stack nav on the settings tab" question [in chat](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/Stack.20nav.20for.20settings.20tab/near/1105309) and decided to not do that, which is fine. 🙂 

I've pushed a new revision, with some mentions of that idea removed from some commit messages. I've also removed the last commit, which would have renamed `SettingsScreen` to `SettingsMenuScreen` or `SettingsInitialScreen`, since there won't be a `SettingsStackScreen` to confuse it with.

**Comment by gnprice** [comment]
Looks good, thanks! Merged.

**Pull request merged**